### PR TITLE
Cleanup all v2 in names in Eventing-controller

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Cleanup all v2 in names in Eventing-controller

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/16310

### CODE REVIEW

File: components/eventing-controller/pkg/backend/utils/utils.go [hunk 1]
```diff
@@ -42,3 +51,51 @@ func ConvertMsgToCE(msg *nats.Msg) (*cev2event.Event, error) {
 	}
 	return &event, nil
 }
+
+func GetExposedURLFromAPIRule(apiRule *apigatewayv1beta1.APIRule, targetURL string) (string, error) {
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/backend/utils/utils.go:55]: Copied from [link](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/pkg/backend/utils/v2/utils.go)

File: components/eventing-controller/pkg/backend/utils/utils.go [hunk 2]
```diff
@@ -42,3 +51,51 @@ func ConvertMsgToCE(msg *nats.Msg) (*cev2event.Event, error) {
...
+		if rule.Path == sURLPath {
+			path = rule.Path
+			break
+		}
+	}
+	return fmt.Sprintf("%s%s%s", scheme, *apiRule.Spec.Host, path), nil
+}
+
+// UpdateSubscriptionStatus updates the status of all Kyma subscriptions on k8s.
+func UpdateSubscriptionStatus(ctx context.Context, dClient dynamic.Interface,
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/backend/utils/utils.go:78]: Copied from [link](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/pkg/backend/utils/v2/utils.go)

File: components/eventing-controller/pkg/backend/utils/utils.go [hunk 3]
```diff
@@ -42,3 +51,51 @@ func ConvertMsgToCE(msg *nats.Msg) (*cev2event.Event, error) {
...
+	_, err = dClient.
+		Resource(eventingv1alpha2.SubscriptionGroupVersionResource()).
+		Namespace(sub.Namespace).
+		UpdateStatus(ctx, unstructuredObj, metav1.UpdateOptions{})
+
+	return err
+}
+
+// LoggerWithSubscription returns a logger with the given subscription (v1alpha2) details.
+func LoggerWithSubscription(log *zap.SugaredLogger,
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/backend/utils/utils.go:93]: Copied from [link](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/pkg/backend/utils/v2/utils.go)

File: components/eventing-controller/testing/matchers.go [hunk 4]
```diff
@@ -157,3 +157,125 @@ func HaveBEBSecretNameAndNamespace(bebSecretName, namespace string) gomegatypes.
 		return s.BEBSecretName == bebSecretName && s.BEBSecretNamespace == namespace
 	}, BeTrue())
 }
+
+func HaveSubscriptionName(name string) gomegatypes.GomegaMatcher {
```
Comment (mfaizanse) [inline components/eventing-controller/testing/matchers.go:161]: Copied from [link](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/testing/v2/matchers.go)

File: components/eventing-controller/testing/test_helpers.go [hunk 5]
```diff
@@ -342,3 +361,498 @@ func PublisherDeploymentNotReadyEvent() corev1.Event {
 		Type:   corev1.EventTypeWarning,
 	}
 }
+
+// NewAPIRule returns a valid APIRule.
+func NewAPIRule(subscription *eventingv1alpha2.Subscription, opts ...APIRuleOption) *apigatewayv1beta1.APIRule {
```
Comment (mfaizanse) [inline components/eventing-controller/testing/test_helpers.go:366]: Copied from [link](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/testing/v2/test_helpers.go)

File: components/eventing-controller/api/v1alpha1/subscription_conversion_unit_test.go [hunk 6]
```diff
@@ -31,57 +30,57 @@ func Test_Conversion(t *testing.T) {
 		{
 			name: "Converting NATS Subscription with empty Status",
 			alpha1Sub: newDefaultSubscription(
-				testingv1.WithEmptyFilter(),
-				testingv1.WithEmptyConfig(),
-				testingv1.WithEmptyStatus(),
+				eventingtesting.WithV1alpha1EmptyFilter(),
```
Comment (k15r) [inline components/eventing-controller/api/v1alpha1/subscription_conversion_unit_test.go:33]: imho here v1Alpha1 should be part of the package as it was before. Is there a reason for this change?

**Review by k15r — APPROVED** [review]

**Pull request merged**